hasjob/search.py

```diff
@@ -95,31 +57,16 @@ def on_models_committed(sender, changes):
 
 @job("hasjob-search")
 def delete_from_index(oblist):
-    ix = index.open_dir(app.config['SEARCH_INDEX_PATH'])
-    writer = ix.writer()
-    for item in oblist:
-        mapping = item.search_mapping()
-        writer.delete_by_term('idref', mapping['idref'])
-    writer.commit()
+    es.bulk([es.delete_op(id=mapping['idref'], doc_type=type_from_idref(mapping['idref'])) for mapping in oblist], index=ES_INDEX)
 
 
 def configure_once():
-    index_path = app.config['SEARCH_INDEX_PATH']
-    if not os.path.exists(index_path):
-        os.mkdir(index_path)
-        ix = index.create_in(index_path, search_schema)
-        writer = ix.writer()
-        # Index everything since this is the first time
-        for model in [models.JobType, models.JobCategory, models.JobPost]:
-            try:
-                for ob in model.query.all():
-                    mapping = ob.search_mapping()
-                    public = mapping.pop('public')
-                    if public:
-                        writer.add_document(**mapping)
-            except ProgrammingError:
-                pass  # The table doesn't exist yet. This is really a new installation.
-        writer.commit()
+    for model in INDEXABLE:
+        records = map(lambda record: record.search_mapping(), model.query.all())
```

Comment:
This will choke on production. It doesn't have enough RAM to do `JobPost.query.all()`.